SwiftReflector/MarshalEngine.cs

```diff
@@ -568,6 +568,11 @@ void AddExtraGenericParameters (FunctionDeclaration wrapperFunc, List<CSBaseExpr
 					callParameters.Add (new CSFunctionCall ("StructMarshal.Marshaler.Metatypeof", false,
 									    argType.Typeof (), ifaceConstrs));
 				}
+			}
+			foreach (GenericDeclaration genDecl in wrapperFunc.Generics) {
+				if (TopLevelFunctionCompiler.GenericDeclarationIsReferencedByGenericClassInParameterList (wrapperFunc, genDecl, typeMapper))
```

Comment:
GenericDeclarationIsReferencedByGenericClassInParameterList

Say that three times fast.

Reply:
Isn't that a humdinger of a name?

Reply:
Supercalifragilisticexpialidocious - 14
GenericDeclarationIsReferencedByGenericClassInParameterList - 21

Confirmed: @stephen-hawley is 50% more awesome than Mary Poppins